Sources/BraintreeVenmo/BTVenmoAppSwitchRedirectURL.swift

```diff
@@ -60,36 +46,48 @@ struct BTVenmoAppSwitchRedirectURL {
         let serializedBraintreeData = try? JSONSerialization.data(withJSONObject: braintreeData)
         let base64EncodedBraintreeData = serializedBraintreeData?.base64EncodedString()
 
-        var appSwitchParameters: [String: Any] = [
-            "x-success": successReturnURL,
-            "x-error": errorReturnURL,
-            "x-cancel": cancelReturnURL,
+        queryParameters = [
+            "x-success": constructRedirectURL(with: returnURLScheme, result: "success"),
+            "x-error": constructRedirectURL(with: returnURLScheme, result: "error"),
+            "x-cancel": constructRedirectURL(with: returnURLScheme, result: "cancel"),
             "x-source": bundleDisplayName,
             "braintree_merchant_id": merchantID,
             "braintree_access_token": accessToken,
             "braintree_environment": environment,
-            "braintree_sdk_data": base64EncodedBraintreeData ?? ""
+            "resource_id": paymentContextID,
+            "braintree_sdk_data": base64EncodedBraintreeData ?? "",
+            "customerClient": "MOBILE_APP"
         ]
+    }
 
-        if let paymentContextID {
-            appSwitchParameters["resource_id"] = paymentContextID
-        }
+    // MARK: - Internal Methods
+
+    func universalLinksURL() -> URL {
+        let universalLinkURL = URL(string: "https://venmo.com/go/checkout")!
+
+        var urlComponent = URLComponents(url: universalLinkURL, resolvingAgainstBaseURL: false)!
+        urlComponent.percentEncodedQuery = BTURLUtils.queryString(from: queryParameters as NSDictionary)
+
+        return urlComponent.url!
```

Comment:
Do we want to leave these force unwraps in? I remember adding these quick for spiking.

Though nothing in here will ever dynamically change, prob best practice to avoid

Reply:
That's a good call, updated here: https://github.com/braintree/braintree_ios/pull/1193/commits/bb18d40c4543a606e03f1bffcac4913b6be97e50. I opted to unwrap it with a guard at the callsite before passing it into `startVenmoFlow`, but we could make the URL optional in `startVenmoFlow` and unwrap it there. I thought it made sense to do earlier in the flow to indicate which URL was invalid vs in `startVenmoFlow` when we may not know which of the two URLs is causing the error. Not opposed to moving it though if you have a strong preference.

Reply:
I like it!